pkgs/node/src/utils.zig

```diff
@@ -0,0 +1,32 @@
+const std = @import("std");
+const Thread = std.Thread;
+const Mutex = Thread.Mutex;
+
+const xev = @import("xev");
+
+pub const EventLoop = struct {
+    loop: xev.Loop,
+    // events from libp2p or other threads will also be pushed on it
+    mutex: Mutex,
+
+    const Self = @This();
+    pub fn init() !Self {
+        const loop = try xev.Loop.init(.{});
+        const mutex = Mutex{};
+
+        return Self{
+            .loop = loop,
+            .mutex = mutex,
```

Comment:
```suggestion
        return Self{
            .loop = loop,
            .mutex = Mutex{},
```

Reply:
You are creating an object and copying it. Not a performance hit, but let's remove copies where we can.

Reply:
what is the better pattren, i don't want to maintain the pointer/reference in the main eventloop struct?

Reply:
there's no pointer here, and they are equivalent. It's just saving one line of code and a copy (that is probably not happening anyway as I'm sure the compiler gets rid of it for us)